client/src/components/App.jsx

```diff
@@ -9,16 +9,10 @@ export default class App extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      product_id: '71697'
+      product_id: new URLSearchParams(window.location.search).get('id')
```

Comment:
this one is more simple and clear. My best takeaway from this code change is. componentDidMount() will render the page after the initial setup.

Reply:
Yes, I noticed that as well, I assumed that it would rerender components on state change but I guess not